### PR TITLE
UPDATED: BuyTokens getting the token amount from a function instead of just multiplication. Useful for making Crowdsales with bonus periods

### LINKED ISSUE #302: Crowdsale token price rate is fixed
Users should be able to customize the price function for Crowdsale contracts.

### PR DESCRIPTION
**Pull request by Perseverance**
I've changed the buyTokens method to get the rate from `getTokenAmount` function. This function can later be overridden to make crowdsales with different rate periods based on some business logic.

This is an update pull request as the previous one was too messy due to conflicts
Fixes #302

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "zeppelin-solidity",
-  "version": "1.5.0",
+  "version": "1.5.1",
```
Comment (AugustoL) [inline package.json]: Why change the version? The version is changed only on each release I think and by the maintainers of the repo, please remove this change.

Comment (Perseverance) [reply package.json]: Done

**Comment by Perseverance** [comment]
The build is failing with "Error: Could not connect to your Ethereum client. Please check that your Ethereum client:" after my commit changing back the lib version. I do not think the error is related to my changes :)

**Comment by AugustoL** [comment]
@Perseverance it is not! Is a problem in travis that happens randomly. Im going to restart it.

**Comment by Perseverance** [comment]
@AugustoL done, should have reverted the first time :) Please squash them if you have such option when merging

File: contracts/crowdsale/Crowdsale.sol [hunk 2]
```diff
@@ -59,12 +59,16 @@ contract Crowdsale {
...
   }
 
-
   // fallback function can be used to buy tokens
   function () external payable {
     buyTokens(msg.sender);
   }
 
+  // Override this function to create logic for periodization
+  function getTokenAmount(uint256 weiAmount) internal constant returns(uint256) {
```
Comment (frangio) [inline contracts/crowdsale/Crowdsale.sol]: Can we change `constant` for `view`? They're aliases but we use `view` throughout the codebase.

Comment (AugustoL) [reply contracts/crowdsale/Crowdsale.sol]: good catch!

**Comment by Perseverance** [comment]
@frangio Changed constant to view ;)

**Comment by Perseverance** [comment]
@frangio Is it good to go now?

**Comment by frangio** [comment]
Thank you @Perseverance!

**Pull request merged**